### PR TITLE
Remove support for idf 3

### PR DESCRIPTION
**Pull request by sorru94**

### CODE REVIEW

File: src/astarte_device.c [hunk 1]
```diff
@@ -77,12 +77,8 @@ static void on_disconnected(astarte_device_handle_t device);
 static void on_incoming(
     astarte_device_handle_t device, char *topic, int topic_len, char *data, int data_len);
 static void on_certificate_error(astarte_device_handle_t device);
-#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 0, 0)
```
Comment (harlem88) [inline src/astarte_device.c:80]: Why did you remove this check? Is it still valid on `4.2`?

File: src/astarte_device.c [hunk 2]
```diff
@@ -1153,20 +1146,11 @@ static void on_certificate_error(astarte_device_handle_t device)
...
 }
 
-#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 0, 0)
 static void mqtt_event_handler(
     void *handler_args, esp_event_base_t base, int32_t event_id, void *event_data)
-#else
-static esp_err_t mqtt_event_handler(esp_mqtt_event_handle_t event)
-#endif
 {
-#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 0, 0)
```
Comment (harlem88) [inline src/astarte_device.c:1163]: same above

File: src/astarte_device.c [hunk 3]
```diff
@@ -1210,7 +1194,4 @@ static esp_err_t mqtt_event_handler(esp_mqtt_event_handle_t event)
             // Handle MQTT_EVENT_ANY introduced in esp-idf 3.2
             break;
     }
-#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)
```
Comment (harlem88) [inline src/astarte_device.c:1213]: same above

**Review by harlem88 — CHANGES_REQUESTED** [review]
I think should be removed idf-3 compatibility also from the Readme.

File: src/astarte_device.c (hunk 1 above)
Comment (sorru94) [reply src/astarte_device.c:80]: Those macros had incorrect arguments, they should have been 
```
#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(4, 0, 0)
```

**Review by harlem88 — COMMENTED** [review]
Add a line to the changelog about `removing the idf-3 support`.

**Review by harlem88 — APPROVED** [review]
